onadata/apps/api/viewsets/organization_profile_viewset.py

```diff
@@ -83,11 +85,22 @@ def retrieve(self, request, *args, **kwargs):
         """Get organization from cache or db"""
         cache_key = get_org_profile_cache_key(request.user, self.get_object())
         cached_org = cache.get(cache_key)
+
         if cached_org:
             return Response(cached_org)
-        response = super().retrieve(request, *args, **kwargs)
-        cache.set(cache_key, response.data)
-        return response
+
+        is_admin = OwnerRole.user_has_role(
```

Comment:
This should probably be in the override function `get_serializer_class()`. Can we not access the user who has made the request there?

Reply:
@ukanga We can't use `self.get_object` inside `get_serializer_class` as it normally results into an infinite recursion